action.yml

```diff
@@ -0,0 +1,34 @@
+name: Shuttle deploy
+description: "Deploy to shuttle"
+
+inputs:
+  shuttle-deploy-key:
+    description: 'The key found at "https://www.shuttle.rs/login"'
+    required: true
+  working-directory:
+    description: "directory which Cargo.toml is under"
+    required: false
+    default: "."
+
+runs:
+  using: "composite"
+  steps:
+    - id: check-repo-is-not-initialized
+      run: echo "::set-output name=remote-url::$( git config --get remote.origin.url )"
+      shell: bash
+    - uses: actions/checkout@v2
+      if: ${{ !contains(steps.check-repo-is-not-initialized.outputs.remote-url, github.repository) }}
+    - name: Install shuttle cli
+      run: |
+        # using cargo-quickinstall because it is faster to get shuttle-cli binary
+        cargo install cargo-quickinstall
+        cargo quickinstall cargo-shuttle
+      shell: bash
+    - name: Log into shuttle
+      run: |
+        cargo shuttle login --api-key ${{ inputs.shuttle-deploy-key }} --allow-dirty

```

Comment:
```suggestion
        cargo shuttle login --api-key ${{ inputs.shuttle-deploy-key }}
```
There should be no unstaged files since the repo was just checked out right?

Reply:
There is a guard in place which means that if the repo has been checked out in the current workflow it doesn't override it so it's not always a fresh checkout. There may be a very niche use case that the user generates some assets in the workflow which aren't committed that need to be sent for the deploy. However I have only ever done that in other languages and because of cargo build scripts I don't think it would be done that way.

Either way the `--allow-dirty` is going to go because I put it on the login command rather than deploy command lol 😆